File: src/PruebaBuilder.java
@Component
public class PruebaBuilder {
  private Prueba borrador;
  private String titulo;
  private String contenido;
  private String imagenUrl;
  private Club club;
  private CategoriaFormativa categoria;

  public PruebaBuilder() {
    this.borrador = new Prueba();
  }

  public PruebaBuilder agregarTitulo(String titulo) {
    this.titulo = titulo;
    return this;
  }

  public PruebaBuilder agregarContenido(String contenido) {
    this.contenido = contenido;
    return this;
  }

  public PruebaBuilder agregarImagenUrl(String imagenUrl) {
    this.imagenUrl = imagenUrl;
    return this;
  }

  public PruebaBuilder agregarClub(Club club) {
    this.club = club;
    return this;
  }

  public PruebaBuilder agregarCategoria(CategoriaFormativa categoria) {
    this.categoria = categoria;
    return this;
  }

  public Prueba construirBorrador() {
    borrador.setTitulo(titulo);
    borrador.setContenido(contenido);
    borrador.setImagenUrl(imagenUrl);
    borrador.setClub(club);
    borrador.setCategoria(categoria);

    return borrador;
  }
}

File: app/model/entities/PruebaBuilder.ts
import { CategoriaFormativa } from "./CategoriaFormativa";
import { Club } from "./Club";
import { Prueba } from "./Prueba";


export class PruebaBuilder {
  private borrador: Prueba;
  private titulo?: string;
  private contenido?: string;
  private imagenUrl?: string;
  private club?: Club;
  private categoria?: CategoriaFormativa;

  constructor() {
    this.borrador = new Prueba();
  }

  agregarTitulo(titulo: string): PruebaBuilder {
    this.titulo = titulo;
    return this;
  }

  agregarContenido(contenido: string): PruebaBuilder {
    this.contenido = contenido;
    return this;
  }

  agregarImagenUrl(imagenUrl: string): PruebaBuilder {
    this.imagenUrl = imagenUrl;
    return this;
  }

  agregarClub(club: Club): PruebaBuilder {
    this.club = club;
    return this;
  }

  agregarCategoria(categoria: CategoriaFormativa): PruebaBuilder {
    this.categoria = categoria;
    return this;
  }

  construirBorrador(): Prueba {
    this.borrador.setTitulo(this.titulo!);
    this.borrador.setContenido(this.contenido!);
    this.borrador.setImagenUrl(this.imagenUrl!);
    this.borrador.setClub(this.club!);
    this.borrador.setCategoria(this.categoria!);

    return this.borrador;
  }
}
